Sources/Sentry/SentryProfiler.mm

```diff
@@ -100,15 +100,20 @@
         threadMetadata[threadID] = metadata;
     }
     if (!backtrace.threadMetadata.name.empty() && metadata[@"name"] == nil) {
-        metadata[@"name"] = [NSString stringWithUTF8String:backtrace.threadMetadata.name.c_str()];
+        const auto name = [NSString stringWithUTF8String:backtrace.threadMetadata.name.c_str()];
+        if (name != nil) {
+            metadata[@"name"] = name;
+        }
     }
     if (backtrace.threadMetadata.priority != -1 && metadata[@"priority"] == nil) {
         metadata[@"priority"] = @(backtrace.threadMetadata.priority);
     }
     if (queueAddress != nil && queueMetadata[queueAddress] == nil
         && backtrace.queueMetadata.label != nullptr) {
-        queueMetadata[queueAddress] =
-            @ { @"label" : [NSString stringWithUTF8String:backtrace.queueMetadata.label->c_str()] };
+        const auto label = [NSString stringWithUTF8String:backtrace.queueMetadata.label->c_str()];
+        if (label != nil) {
+            queueMetadata[queueAddress] = @ { @"label" : label };
```

Comment:
this is legit 👍🏽

Reply:
This is the only dictionary literal in processBacktrace which was the blamed frame in the issue screenshot, so is the likely candidate actually needing this. I'll just remove all the others.

Reply:
`h`: @armcknight, I think this won't work. As pointed out above, `stringWithUTF8String` is not supposed to return `nil`. Maybe the `label->c_str()` isn't returning a proper NULL-terminated C array of bytes in UTF8 encoding.

Reply:
I'm afraid if we don't think this check will be useful, then we will have to close this PR and go back to the drawing board. Perhaps there is something else going on and the crash reporter is errantly blaming this frame.

`std::string::c_str()` will never return `NULL` as long as the reference is not `NULL`, which we already guard against in the `if` condition right above these lines: https://github.com/getsentry/sentry-cocoa/blob/034be1c2ce49e44fc857dbda11657371e7cac85e/Sources/Sentry/SentryProfiler.mm#L108-L109

Reply:
I force pushed a new commit so that all the other nil checks in the PR are removed, and changed this one to check the return value of `c_str()` is `!= nullptr` before passing it to `stringWithUTF8String`: https://github.com/getsentry/sentry-cocoa/pull/2995/files#diff-9d7d24f33a2554811f36d8236cdd5b9b2c7c2786b12be31e4a05a3934193441cR110-R114

```
const auto labelCString = backtrace.queueMetadata.label->c_str();
if (labelCString != nullptr) {
    queueMetadata[queueAddress] =
        @ { @"label" : [NSString stringWithUTF8String:labelCString] };
}
```

but i'm not sure this is doing anything actually useful.

Reply:
But then I think the change doesn't make sense as you pointed out and we have to go back to the drawing board 🤔

Reply:
So should we close this PR @indragiek and @philipphofmann ?

Reply:
I agree with closing it. Although we can't merge it, it was still worth the effort as it helped us to understand the underlying problem.